superset/reports/notifications/base.py

```diff
@@ -21,11 +21,15 @@
 import pandas as pd
 
 from superset.reports.models import ReportRecipients, ReportRecipientType
+from superset.utils.core import HeaderDataType
 
 
 @dataclass
 class NotificationContent:
     name: str
+    header_data: Optional[
+        HeaderDataType
+    ] = None  # this is optional to account for error states
```

Comment:
can you explain why this is optional?

Reply:
Yeah, we also create notification_content when there is an error, should we be pushing header data into those cases as well? I assumed we wouldn't be but happy to make it not optional, and create header data in those instances as well.